helpers/mysql/lib/opentelemetry/helpers/mysql.rb

```diff
@@ -40,7 +40,7 @@ class MySQL
       # Ignore query names that might appear in comments prepended to the
       # statement.
       PREPENDED_COMMENTS_REGEX = %r{/\*(?:[^/]|/[^*])*?(?:\*/|/\*.*)}m
-      QUERY_NAME_REGEX = Regexp.new("(?:#{PREPENDED_COMMENTS_REGEX})?.*\\b(#{QUERY_NAMES.join('|')})\\b.*", Regexp::IGNORECASE)
+      QUERY_NAME_REGEX = Regexp.new("(?:#{PREPENDED_COMMENTS_REGEX})?.*?\\b(#{QUERY_NAMES.join('|')})\\b.*", Regexp::IGNORECASE)
```

Comment:
This seems overly permissive. I think we want to:
1. match against the start of the SQL
2. accept initial whitespace
3. allow arbitrary whitespace (only) between any comment and the query name.

From informal testing, this seems to meet those requirements:
```suggestion
      QUERY_NAME_REGEX = Regexp.new("^\s*(?:#{PREPENDED_COMMENTS_REGEX})?\s*\\b(#{QUERY_NAMES.join('|')})\\b.*", Regexp::IGNORECASE)
```